tests/ctst/steps/iam-policies/common.ts

```diff
@@ -0,0 +1,212 @@
+            checkResponse: this.putObjectRetentionResponseCode,
+        },
+        {
+            API: 'GetObjectVersionLegalHold',
+            checkResponse: this.getObjectVersionLegalHoldResponseCode,
+        },
+        {
+            API: 'PutObjectVersionLegalHold',
+            checkResponse: this.putObjectLegalHoldTaggingResponseCode,
+        },
+        {
+            API: 'GetObjectVersionTagging',
+            checkResponse: this.getObjectVersionTaggingResponseCode,
+        },
+        {
+            API: 'DeleteObjectVersionTagging',
+            checkResponse: this.deleteObjectVersionTaggingResponseCode,
+        },
+        {
+            API: 'PutObjectVersionTagging',
+            checkResponse: this.putObjectTaggingResponseCode,
+        },
+        {
+            API: 'GetObjectVersionAcl',
+            checkResponse: this.getObjectAclResponseCode,
+        },
+        {
+            API: 'PutObjectVersionAcl',
+            checkResponse: this.putObjectAclResponseCode,
+        },
+        {
+            API: 'GetBucketTagging',
+            checkResponse: this.getBucketTaggingResponseCode,
+        },
+        {
+            API: 'PutBucketTagging',
+            checkResponse: this.putBucketTaggingResponseCode,
+        },
+        {
+            API: 'DeleteBucketTagging',
+            checkResponse: this.deleteBucketTaggingResponseCode,
+        },
+    ];
+    
+    let userCredentials;
+    if ([EntityType.IAM_USER, EntityType.ACCOUNT].includes(this.saved.type)) {
+        userCredentials = this.parameters.IAMSession;
+    } else {
+        userCredentials = this.parameters.AssumedSession;
+    }
+    console.log("userCredentialsuserCredentialsuserCredentials");
+    console.log(userCredentials);
+    console.log("thisthisthis");
+    console.log(this);
+    if (ifS3Standard === 'notS3Standard'){
+        this.saved.ifS3Standard = false;
+        const api = testAPIs.find(api => api.API === action);
+        // Handle Metadatasearch special case
+        if (action === 'MetadataSearch') {
+            this.result = await this.metadataSearchResponseCode(userCredentials, this.saved.bucketName);
+        }
+        // Handle other cases
+        else if (api) {
+            this.result = await api.checkResponse();
+        }
+    } else {
+        this.resetCommand();
+        this.saved.ifS3Standard = true;
+        this.addCommandParameter({bucket: this.saved.bucketName});
+        if (this.saved.objectName && this.saved.objectName != '') {
+            this.addCommandParameter({key: this.saved.objectName});
+        }
+        this.result = await this.restoreObjectResponseCode();
+    }
+});
+
+Then('the user should be able to perform successfully', function () {
+    this.endForType();
+    if (this.saved.ifS3Standard) {
+        assert.strictEqual(this.result?.err?.includes("AccessDenied"), false);
+    } else {
+        assert.strictEqual(this.result?.statusCode, 200);
+    }
+});
```

Comment:
Can we check instead that we got a valid response? In case the API crashes with a given identity, we won't get an AccessDenied, but not a valid response either...

Reply:
we did consider this part, some API calls just can't return a valid response, like restoreObject, like putObjectRetention, the condition required to get a successfully response for these calls  is not easy to provide, as we are not testing the functionalities of these APIs, so we are thinking of specifying the expected error that is returned for each API call. like for restoreObject it's `invalidObjectState`, for putObjectRetention, it's `InvalidRequest, bucket is missing object lock configuration`. we only accept the expected error returned, if other error returns, we just fail the tests